### PR TITLE
Cleanup & Fix: Parsers, Equip, and etc

### PR DESCRIPTION
**Pull request by rhydev**
- Removed ConfigurationManager from GameDataParser and just use static class for paths instead
- Resources folders in GameDataParser and Maple2Storage now automatically get created if missing
- If missing xml or exported pack files now displays an informative message in console
- Changed path handling in GameDataParser and Maple2Server to no longer use relative path (program can be started from any working directory now)
- Cleaned up file path references
- Rewrote item parser to no longer use XmlReader
- Added IsTwoHand bool to item to detect 2h weapons
- Renamed SlotMax to StackLimit in Item
- Added Cosmetic equipment to Player
- Fixed ItemParser item box parsing to ignore locales other than NA in NA folder
- Fixed item equipping and unequipping to properly unequip items when going from 1h to 2h or 2h to 1h
- Fixed item unequipping not returning item to proper slot (first available slot when unequip, replace old item slot when equip + unequip)
- Fixed GuildMetadataExport to properly run GuildParser instead of PrestigeParser
- Fixed PacketWriter WriteString
- Cleaned up formatting (dotnet-format)
- Fixed ItemUseHelper to include job wildcard (Job.NONE)
- Fixed bad logic in InventoryController Update

### CODE REVIEW

File: GameDataParser/Parsers/ScriptParser.cs [hunk 1]
```diff
@@ -249,14 +249,14 @@ public static List<ScriptMetadata> ParseQuest(MemoryMappedFile m2dFile, IEnumera
...
+            using (FileStream writeStream = File.Create($"{Paths.OUTPUT}/ms2-script-metadata"))
             {
                 Serializer.Serialize(writeStream, entities);
             }
-            using (FileStream readStream = File.OpenRead(VariableDefines.OUTPUT + "ms2-script-metadata"))
+            using (FileStream readStream = File.OpenRead($"{Paths.OUTPUT}/ms2-script-metadata"))
             {
             }
-            Console.WriteLine("Successfully parsed script metadata!");
+            Console.WriteLine("\rSuccessfully parsed script metadata!");
```
Comment (AlanMorel) [inline GameDataParser/Parsers/ScriptParser.cs:259]: `Environment.NewLine` is preferable here

https://docs.microsoft.com/en-us/dotnet/api/system.environment.newline?view=net-5.0

File: GameDataParser/Program.cs [hunk 2]
```diff
@@ -20,51 +20,70 @@ private static void Main()
...
+                insigniaThread.Join();
+                prestigeThread.Join();
+                expThread.Join();
+                questThread.Join();
+                scriptThread.Join();
+                guildThread.Join();
+
+                spinner.Stop();
+
+                Console.WriteLine("\r\nFinished parsing.");
```
Comment (AlanMorel) [inline GameDataParser/Program.cs:77]: Same here, and anywhere else you're trying to start on a new line

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Review by angelotadeucci — CHANGES_REQUESTED** [review]

**Review by Sparkymod — COMMENTED** [review]
Looks good

File: MapleServer2/Data/Static/ItemMetadataStorage.cs [hunk 3]
```diff
@@ -53,9 +54,14 @@ public static int GetRarity(int itemId)
...
         }
 
-        public static int GetSlotMax(int itemId)
+        public static int GetStackLimit(int itemId)
         {
-            return map.GetValueOrDefault(itemId).SlotMax;
+            return map.GetValueOrDefault(itemId).StackLimit;
+        }
+
+        public static bool GetIsTwoHand(int itemId)
```
[the remark this answers is not in the document]
Comment (rhydev) [reply MapleServer2/Data/Static/ItemMetadataStorage.cs:62]: Well it's a Getter so it should be Get, if we were just accessing a property then it'd be just IsTwoHand

**Pull request merged**